daml-assistant/daml-helper/test/DA/Daml/Helper/Test/Ledger.hs

```diff
@@ -1,24 +1,41 @@
 -- Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
 -- SPDX-License-Identifier: Apache-2.0
 
-module DA.Daml.Helper.Test.Ledger (main) where
+module DA.Daml.Helper.Test.Ledger
+  ( main
+  ) where
 
+import qualified "zip-archive" Codec.Archive.Zip as Zip
 import DA.Bazel.Runfiles
+import qualified DA.Daml.LF.Ast.Base as LF
+import qualified DA.Daml.LF.Proto3.Archive as LF
+import qualified DA.Daml.LF.Reader as LF
 import DA.Ledger.Services.PartyManagementService (PartyDetails(..))
 import DA.Ledger.Types (Party(..))
 import DA.Test.HttpJson
 import DA.Test.Sandbox
+import qualified Data.ByteString.Lazy as BSL
+import qualified Data.Text as T
 import System.Environment.Blank
 import System.FilePath
+import System.IO.Extra
 import System.Process
 import Test.Tasty
 import Test.Tasty.HUnit
 
+readDarMainPackageId :: FilePath -> IO String
```

Comment:
The `inspect-dar` command has to solve a similar problem. Maybe there's already a function for this purpose?

Reply:
I exposed one in the inspect-dar code.